src/shogun/regression/GLM.cpp

```diff
@@ -0,0 +1,38 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/regression/GLM.h>
+#include <utility>
+using namespace shogun;
+
+GLM::GLM() : LinearMachine()
+{
+	init();
+}
+
+void GLM::init()
+{
+	SG_ADD(&m_alpha, "alpha", "alpha parameter", ParameterProperties::HYPER);
+	SG_ADD(&m_lambda, "lambda", "lambda parameter", ParameterProperties::HYPER);
+	SG_ADD(
+	    (std::shared_ptr<SGObject>*)&m_descend_updater, "descend_updater",
+	    "Descend Updater used for updating weights",
+	    ParameterProperties::SETTING);
+	SG_ADD(&m_family, "family", "family used", ParameterProperties::SETTING);
+	SG_ADD(
+	    &m_linkfn, "linkfn", "Link function used",
+	    ParameterProperties::SETTING);
+}
+GLM::GLM(
+    const std::shared_ptr<DescendUpdater>& descend_updater, Family family,
```

Comment:
Another comment as I saw a concversation in irc re swig and the fact that the descend updater is not exposed yet. I would not worry about swig and the interfaces for now. Let's focus on the c++ code for now, possibly with unit tests.
The next step here (after incorporating the feedback you got) is to implement the gradient of the log-likelihood, and the next thing then is passing that to a descent updater. For the latter, you can start with plain old gradient descent, no need to be more fancy (for now).